src/init/application/application.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as chalk from 'chalk';
+import * as path from 'path';
+import {Base, IPromptOptions} from 'yeoman-generator';
+
+export class ApplicationGenerator extends Base {
+
+  props: any;
+
+  constructor(args: string | string[], options: any) {
+    super(args, options);
+    this.sourceRoot(path.join(__dirname, 'templates'));
+  }
+
+  initializing() {
+    // Yeoman replaces dashes with spaces. We want dashes.
+    this.appname = this.appname.replace(/\s+/g, '-');
+  }
+
+  prompting() {
+    var _this = this;
+
+    var prompts = [
+      {
+        name: 'name',
+        type: 'input',
+        message: `Application name (${this.appname})`,
+        default() {
```

Comment:
nit: function is unnecessary here. if possible all defaults should be strings, or functions.